### PR TITLE
Add Dev secret modal on Dust Apps list screen in global vault

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

This PR adds a new "Dev secrets" button with a new modal to manage Devs secrets from the new page to list Dust apps from the Global vault. 
I reused the existing design to write at little code as possible and share the code with the old version. 

I'll show the full App experience to Ed tomorrow and apply his feedback. 

### List view: 
<kbd>
Screenshot 2024-08-29 at 18 31 56
</kbd>

### Modal: 
<kbd>
Screenshot 2024-08-29 at 18 25 03
</kbd>

## Risk

/ 

## Deploy Plan

Deploy front.

### CODE REVIEW

File: front/components/app/AppSecretsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export const AppSecretsModal = ({
+  owner,
+  isOpen,
+  setIsOpen,
+}: {
+  owner: WorkspaceType;
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+}) => {
+  return (
```
Comment (flvndvd) [inline front/components/app/AppSecretsModal.tsx:15]: For once, we can bundle the button in this component 🙏

**Comment by albandum** [comment]
Where are the API keys ? 😅

**Comment by PopDaph** [comment]
@albandum there's an in progress card for it, they will be on the admin menu.

**Review by flvndvd — APPROVED** [review]
🔥 🔥

**Pull request merged**